app/models/patient.rb

```diff
@@ -51,18 +51,35 @@ class Patient < ApplicationRecord
 
   attribute :call_result, :string
 
-  #
-  # Note: This scope expects a join(:blood_pressures) to exist.
-  # For eg, Patient.joins(:blood_pressures).follow_ups(:month).
-  #
-  # It doesn't include the join in this scope to play well with certain parent scopes (eg. Facility).
-  # Parent scopes might auto add this join and the final query would end up with unnecessary joins, affecting perf.
-  #
+  scope :diabetes_only, -> { joins(:medical_history).merge(MedicalHistory.diabetes_yes) }
+  scope :hypertension_only, -> { joins(:medical_history).merge(MedicalHistory.hypertension_yes) }
```

Comment:
Makes sense, I think `with_hypertension` works. I don't want to use `diagnosed_with_hypertension` because it can be confused with the field in MedicalHistory called `diagnosed_with_hypertension`.